### ISSUE TITLE
SwiftFormat inserts self where local variable shadows class instance method

### ISSUE THREAD

**Issue by Panajev** [nicklockwood/SwiftFormat#259]
Hello :),

We have noticed that SwiftFormat handles the following case
```swift
    var lastNonLoggedInRoute: NativeHistory? {
        for route in self.history.reversed() where route.isLogin == false {
            return route
        }
        return nil
    }
```
incorrectly when adding **self.** as a formatting rule. 
Since the class this code snippet comes from has a method that is named route, I think SwiftFormat may be thinking that route in the code snippet is the 'route method' and not the local variable route and will transform the code snippet into:
```swift
    var lastNonLoggedInRoute: NativeHistory? {
        for route in self.history.reversed() where self.route.isLogin == false {
            return self.route
        }
        return nil
    }
```

Besides renaming the variable, what is the proper long-term course of action? 

1.) Should SwiftFormat emit an error there and refuse to do the transformation?

2.) Should SwiftFormat avoid to apply the transformation in which case recognising the fact route is not the method, but the local variable?

I think the first option is the safest and easiest as it avoids introducing mistakes in users' code and allows the tool autotransformation to be trusted more without double-checking its output.

Kind Regards,

Goffredo

**Comment by nicklockwood** [comment]
@Panajev this sounds like a bug in swiftFormat, so the correct course of action for me is to fix it! 

For you, the correct course is probably to use comments to temporarily disable the rule for inside the loop:

    // swiftformat:disable redundantSelf
    return route
    // swiftformat:enable redundantSelf

**Comment by Panajev** [comment]
Cheers Nick, for sure fixing it is the bestest course of action :)!

Naively, I would think that such situation, shadowing a method with a local variable should be considered a kind of code smell if it can be avoided, but anything that can be done to ensure I do not have my developers fear this autocorrect the better.

I am quite thrilled with both SwiftLint and SwiftFormat working together and agreeing :) (albeit I would like colon in a type declaration or a method parameter label to be treated differently than how colon is used on a dictionary [key:value]  vs   var a: MyClass... but this is a topic for another thread :)).

**Comment by nicklockwood** [comment]
@Panajev detecting code smells is more the domain of SwiftLint. As long as the code compiles, SwiftFormat should accept it too.

**Comment by Panajev** [comment]
Reasonable :).

**Comment by nicklockwood** [comment]
Fixed in version 0.33.8
